framework/test/unit/node/consensus/certificate_generation/commit_pool.spec.ts

```diff
@@ -89,13 +95,373 @@ describe('CommitPool', () => {
+					certificateSignature: Buffer.alloc(0),
+					height: 1030,
+				},
+			});
+
+			certificate = computeCertificateFromBlockHeader(blockHeader);
+
+			privateKey = generatePrivateKey(getRandomBytes(32));
+			publicKey = getPublicKeyFromPrivateKey(privateKey);
+			signature = signCertificate(privateKey, networkIdentifier, certificate);
+
+			commit = {
+				blockID: blockHeader.id,
+				certificateSignature: signature,
+				height: blockHeader.height,
+				validatorAddress: blockHeader.generatorAddress,
+			};
+
+			chain.finalizedHeight = commit.height - 1;
+
+			weights = Array.from({ length: 103 }, _ => 1);
+			validators = weights.map(weight => ({
+				address: getRandomBytes(20),
+				bftWeight: BigInt(weight),
+			}));
+			// Single commit owner must be an active validator
+			validators[0] = {
+				address: commit.validatorAddress,
+				bftWeight: BigInt(1),
+			};
+
+			when(chain.dataAccess.getBlockHeaderByHeight)
+				.calledWith(commit.height)
+				.mockReturnValue(blockHeader);
+
+			bftAPI.getBFTHeights.mockReturnValue({
+				maxHeightCertified,
+				maxHeightPrecommitted,
+			});
+
+			when(bftAPI.getBFTParameters).calledWith(apiContext, commit.height).mockReturnValue({
+				certificateThreshold: threshold,
+				validators,
+			});
+
+			when(validatorsAPI.getValidatorAccount)
+				.calledWith(apiContext, commit.validatorAddress)
+				.mockReturnValue({ blsKey: publicKey });
+
+			bftAPI.existBFTParameters.mockReturnValue(true);
+
+			when(getBlockHeaderByHeight)
+				.calledWith(chain.finalizedHeight)
+				.mockReturnValue(blockHeaderOfFinalizedHeight);
+		});
+
+		it('should validate single commit successfully', async () => {
+			const isCommitValid = await commitPool.validateCommit(apiContext, commit);
+
+			expect(isCommitValid).toBeTrue();
+		});
+
+		it('should return false when single commit block id is not equal to chain block id at same height', async () => {
+			when(chain.dataAccess.getBlockHeaderByHeight)
+				.calledWith(commit.height)
+				.mockReturnValue(createFakeBlockHeader({ id: getRandomBytes(32) }));
+
+			const isCommitValid = await commitPool.validateCommit(apiContext, commit);
+
+			expect(isCommitValid).toBeFalse();
+		});
+
+		it('should return false when single commit exists in gossiped commits but not in non-gossipped commits', async () => {
+			commitPool['_gossipedCommits'].set(commit.height, [commit]);
+
+			const isCommitValid = await commitPool.validateCommit(apiContext, commit);
+
+			expect(isCommitValid).toBeFalse();
+		});
+
+		it('should return false when single commit exists in non-gossiped commits but not in gossipped commits', async () => {
+			commitPool['_nonGossipedCommits'].set(commit.height, [commit]);
+
+			const isCommitValid = await commitPool.validateCommit(apiContext, commit);
+
+			expect(isCommitValid).toBeFalse();
+		});
+
+		it('should return false when single commit exists both in gossiped commits and in non-gossipped commits', async () => {
```

Comment:
This should never be the case when a commit exists in both gossiped and non-gossiped as we always make this sure while adding to commit pool

Reply:
Will remove.

Reply:
done